mtags/src/main/scala-2/scala/meta/internal/pc/ExtractMethodProvider.scala

```diff
@@ -0,0 +1,138 @@
+package scala.meta.internal.pc
+
+import scala.reflect.internal.util.Position
+
+import scala.meta.internal.mtags.MtagsEnrichments._
+import scala.meta.internal.pc.ExtractMethodUtils
+import scala.meta.pc.OffsetParams
+import scala.meta.pc.RangeParams
+
+import org.eclipse.{lsp4j => l}
+
+final class ExtractMethodProvider(
+    val compiler: MetalsGlobal,
+    range: RangeParams,
+    extractionPos: OffsetParams
+) extends ExtractMethodUtils {
+  import compiler._
+  def extractMethod: List[l.TextEdit] = {
+    val text = range.text()
+    val params = range.toOffset
```

Comment:
`range` should also have `offset` method. Do we need to convert it and have the additional convertion method?